### PR TITLE
Add support for filtering building of packages based on Linux variants.

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
When building native Linux packages we need the ability to build the
packages specific to the distro on which we're building.  I have
overloaded the OSGroup property to specify the Linux distro to which
a package build belongs so the other distros get filtered out.

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
@weshaggard @ericstj @chcosta here's the change for filtering linux package builds.

File: src/Native/pkg/runtime.native.System.IO.Compression/runtime.native.System.IO.Compression.builds [hunk 1]
```diff
@@ -2,21 +2,16 @@
 <Project ToolsVersion="12.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.props))\dir.props" />
   <ItemGroup>
-    <Project Include="runtime.native.System.IO.Compression.pkgproj"/>
```
Comment (weshaggard) [inline src/Native/pkg/runtime.native.System.IO.Compression/runtime.native.System.IO.Compression.builds]: Why did you remove the building of the root level pkgproj's? I would expect them to still be necessary.

File: src/packages.builds [hunk 2]
```diff
@@ -5,11 +5,13 @@
   <PropertyGroup>
     <AdditionalProperties>BuildPackageLibraryReferences=false</AdditionalProperties>
     <PackageReportDir Condition="'$(PackageReportDir)' == ''">$(BinDir)pkg/reports/</PackageReportDir>
+    <BuildAllOSGroups Condition="'$(OSGroup)' != '' OR '$(FilterToOSGroup)' != ''">false</BuildAllOSGroups>
```
Comment (weshaggard) [inline src/packages.builds]: OSGoup is never going to be empty as it is always defaulted to AnyOS in dir.props if it is empty so I don't think that part of the condition is necessary.

File: src/Native/pkg/runtime.native.System.IO.Compression/runtime.native.System.IO.Compression.builds (hunk 1 above)
Comment (jhendrixMSFT) [inline src/Native/pkg/runtime.native.System.IO.Compression/runtime.native.System.IO.Compression.builds]: These packages don't contain any content.  If they are supposed to then that's a different bug and I will add them back.

File: src/packages.builds (hunk 2 above)
Comment (jhendrixMSFT) [inline src/packages.builds]: Will clean up (it was from a copy/paste).

File: src/Native/pkg/runtime.native.System.IO.Compression/runtime.native.System.IO.Compression.builds (hunk 1 above)
Comment (jhendrixMSFT) [inline src/Native/pkg/runtime.native.System.IO.Compression/runtime.native.System.IO.Compression.builds]: Chatted with @chcosta and he says we need them so will add them back.  I still think there's a bug here though with respect to their content (absence thereof).

**Comment by ellismg** [comment]
I think long term having names like: "rhel" and "ubuntu" and "debian" is going to be problematic (as we want to add more versions of these distros, which we'll want to do for Ubuntu during the RTM milestone).  Could we do something like "rhel.7", "ubuntu.14.04" and "debian.8" to more closely match the RIDs of the packages we will be creating?

File: src/Native/pkg/runtime.native.System.IO.Compression/runtime.native.System.IO.Compression.builds (hunk 1 above)
Comment (ericstj) [inline src/Native/pkg/runtime.native.System.IO.Compression/runtime.native.System.IO.Compression.builds]: That's intentional.  They are just markers that can be referenced by other non-distro-specific package.  The runtime.json in the lineup package is what brings in the platform specific implementation package.

File: build-packages.sh [hunk 3]
```diff
@@ -11,8 +11,36 @@ working_tree_root="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 build_packages_log=$working_tree_root/build-packages.log
 binclashlog=$working_tree_root/binclash.log
 binclashloggerdll=$working_tree_root/Tools/Microsoft.DotNet.Build.Tasks.dll
+RuntimeOS=ubuntu
 
-options="/m /nologo /v:minimal /clp:Summary /flp:v=detailed;Append;LogFile=$build_packages_log /l:BinClashLogger,$binclashloggerdll;LogFile=$binclashlog"
+# Use uname to determine what the OS is.
+OSName=$(uname -s)
+case $OSName in
```
Comment (weshaggard) [inline build-packages.sh]: Somewhat orthogonal but we seem to be duplicating this logic in various .sh files we should probably figure out a way to share this logic. @mellinoe made a suggestion to be on friday which I think kind of makes some since for the dev workflow logic. @maririos we should consider adding an msbuild.cmd/sh script that captures a lot of this common logic (i.e. OSGroup selection, common logging, init-tools, etc) and have all these other wrappers simply call that msbuild script with the extra arguments unique to that script. That will allow us to encapsulate a lot of this logic and also provide a way for us to scope the need for having to add msbuild to the path.

File: build-packages.sh [hunk 4]
```diff
@@ -11,8 +11,36 @@ working_tree_root="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 build_packages_log=$working_tree_root/build-packages.log
 binclashlog=$working_tree_root/binclash.log
 binclashloggerdll=$working_tree_root/Tools/Microsoft.DotNet.Build.Tasks.dll
+RuntimeOS=ubuntu
```
Comment (weshaggard) [inline build-packages.sh]: Any reason you are defaulting to ubuntu?

File: dir.traversal.targets [hunk 5]
```diff
@@ -32,11 +32,14 @@
         <AdditionalProperties Condition="'%(Project.InputOSGroup)' != ''">InputOSGroup=%(Project.InputOSGroup);%(Project.AdditionalProperties)</AdditionalProperties>
       </Project>
       <Project>
+        <AdditionalProperties Condition="'%(Project.BuildAllOSGroups)' != ''">BuildAllOSGroups=%(Project.BuildAllOSGroups);%(Project.AdditionalProperties)</AdditionalProperties>
```
Comment (weshaggard) [inline dir.traversal.targets]: FYI @mellinoe is also making this same change so I suspect which ever one of you guys gets in first the other will need to merge this out.

**Comment by weshaggard** [comment]
I agree with @ellismg that given we are targeting these runtime packages at specific versions we should encode the version in the name so we don't add a bunch of debt for future us. 

@maririos should follow-up on the dev workflow script aspects called out. Otherwise the change looks good to me.

File: build-packages.sh (hunk 3 above)
Comment (maririos) [inline build-packages.sh]: @weshaggard I agree on your proposal. I opened the issue #7652 to keep track of it.

**Comment by maririos** [comment]
I created Issue #7652 to make the change in the dev workflow
